src/js/calculator.js

```diff
@@ -0,0 +1,84 @@
+import { getOperations } from "./utils/operations.js";
+
+export const INITIAL_STATE = {
+  operation: undefined,
+  firstTerm: undefined,
+  secondTerm: undefined,
+};
+
+export class Calculator {
+  state = { ...INITIAL_STATE };
+
+  constructor($app) {
+    this.$app = $app;
+    this.$total = $app.querySelector("#total");
+  }
+
+  setState(key, value) {
+    if (this.state.hasOwnProperty(key)) {
+      this.state[key] = value;
+      this.render();
+    } else {
+      throw new Error("Invalid state key");
+    }
+  }
+
+  render = () => {
+    const { firstTerm = "", secondTerm = "", operation = "" } = this.state;
+    const text = `${firstTerm}${operation}${secondTerm}`;
+    this.$total.innerText = text.trim().length === 0 ? "0" : text;
+  };
```

Comment:
![image](https://user-images.githubusercontent.com/18749057/177042376-b281f8eb-f6a0-418e-96db-2e6526bbba29.png)
빈 문자열에 Number를 씌우면 0이 된답니다!

그리고 render가 하는 역할이 너무 많은 것 같아요.
이렇게 나눠주면 어떨까요?
```suggestion
  get result = () => {
    const { firstTerm = "", secondTerm = "", operation = "" } = this.state;
    return Number(`${firstTerm}${operation}${secondTerm}`);
  }
  
  render = () => {
    this.$total.innerText = this.result;
  };
```

Reply:
오 꿀팁 감사합니다! 확실히 문자열을 만드는 코드와 렌더링 하는 코드는 분리하는 게 좋을 듯합니다.